enhancements/cloud-integration/aws/aws-sts-support.md

```diff
@@ -0,0 +1,190 @@
+---
+title: aws-sts-support
+authors:
+  - "@dgoodwin"
+  - "@joelddiaz"
+reviewers:
+  - "@joelddiaz"
+  - "@sjenning"
+  - "@derekwaynecarr"
+  - "@sdodson"
+  - "@abhinavdahiya"
+approvers:
+  - "@derekwaynecarr"
+creation-date: 2020-11-13
+last-updated: 2020-11-13
+status: implementable
+see-also:
+  - "/enhancements/cloud-integration/aws/aws-pod-identity.md"
+---
+
+# AWS STS Token Support
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Add support for using time limited AWS Security Token Service credentials in OpenShift Installation and all core CVO managed operators. This enhancement will avoid the need to use long lived traditional AWS access key ID / secret access key credentials at any point during the lifecycle of the cluster. Support will require manual setup, but once complete tokens will be automatically rotated and made available to affected OpenShift operators.
+
+## Motivation
+
+Better align with AWS best practices in using short, time limited credentials within our operators.
+
+### Goals
+
+List the specific goals of the proposal. How will we know that this has succeeded?
+
+  * OpenShift can be installed with an STS credential.
+  * All core CVO managed OpenShift operators using cloud credentials (approximately 5) can now use STS tokens rotated on a regular basis.
+  * OpenShift can be installed without requiring, generating or using any long lived credentials. (aws_access_key_id / aws_secret_access_key)
+
+### Non-Goals
+
+  * Changing the default install method on AWS. We will continue to use long lived credentials by default to preserve the simplest path to success possible. In future, if we proceed to later phases of STS support where CCO automates the setup fully, we may be able to make this the default.
+  * Leveraging the Cloud Credential Operator to automate this setup required for using STS.
+  * Upgrading pre-existing clusters to use STS. For now this is for new installs only.
+
+## Proposal
+
+### Implementation Details/Notes/Constraints
+
+#### Cloud Credential Operator Changes
+
+The Cloud Credential Operator will be modified to include a new `credentials` Secret key in the target secrets for each `CredentialsRequest`. This `credentials` Secret key will contain a standard AWS credentials file:
+
```

Comment:
Might show the actual secret content, with the base64 encoded `credentials` key and then show that the config file that value decodes to contains the same user/key as the secret.  Just for clarity.